### PR TITLE
Added Spring Annotations

### PR DESCRIPTION
**Pull request by vremere-a**

### CODE REVIEW

File: src/main/java/com/dev/cinema/config/AppConfig.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        properties.put("hibernate.hbm2ddl.auto", environment.getProperty("hibernate.hbm2ddl.auto"));
+        properties.put("hibernate.format_sql", environment.getProperty("hibernate.format_sql"));
+        factoryBean.setHibernateProperties(properties);
+        factoryBean.setAnnotatedClasses(User.class);
+        factoryBean.setAnnotatedClasses(Ticket.class);
+        factoryBean.setAnnotatedClasses(ShoppingCart.class);
+        factoryBean.setAnnotatedClasses(Order.class);
+        factoryBean.setAnnotatedClasses(MovieSession.class);
+        factoryBean.setAnnotatedClasses(Movie.class);
+        factoryBean.setAnnotatedClasses(CinemaHall.class);
```
Comment (aleksandr-hrankin) [inline src/main/java/com/dev/cinema/config/AppConfig.java]: Посмотри checklist)

File: src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java [hunk 2]
```diff
@@ -2,24 +2,33 @@
...
 import org.hibernate.Transaction;
 import org.hibernate.query.Query;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
 
 @Log4j
-@Dao
+@Repository
 public class CinemaHallDaoImpl implements CinemaHallDao {
+
```
Comment (aleksandr-hrankin) [inline src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java]: empty line

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java [hunk 3]
```diff
@@ -14,14 +12,25 @@
...
+import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
 
 @Log4j
-@Dao
+@Repository
 public class MovieSessionDaoImpl implements MovieSessionDao {
+
```
Comment (aleksandr-hrankin) [inline src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java]: empty line

File: src/main/java/com/dev/cinema/security/implementations/AuthenticationServiceImpl.java [hunk 4]
```diff
@@ -1,22 +1,26 @@
...
 @Service
 public class AuthenticationServiceImpl implements AuthenticationService {
-    @Inject
-    private UserService userService;
+    private final UserService userService;
+    private final ShoppingCartService shoppingCartService;
 
-    @Inject
-    private ShoppingCartService shoppingCartService;
+    @Autowired
```
Comment (aleksandr-hrankin) [inline src/main/java/com/dev/cinema/security/implementations/AuthenticationServiceImpl.java:18]: Если у нас один конструктор, то эта аннотация не обязательна

File: src/main/java/com/dev/cinema/config/AppConfig.java (hunk 1 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/config/AppConfig.java]: точно. забыл. спасибо

File: src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java (hunk 3 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/dao/impl/MovieSessionDaoImpl.java]: done

File: src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java (hunk 2 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/dao/impl/CinemaHallDaoImpl.java]: done

File: src/main/java/com/dev/cinema/security/implementations/AuthenticationServiceImpl.java (hunk 4 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/security/implementations/AuthenticationServiceImpl.java:18]: знаю) пусть будет🙃

**Review by aleksandr-hrankin — APPROVED** [review]

File: pom.xml [hunk 5]
```diff
@@ -39,16 +45,60 @@
...
+        </dependency>
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>servlet-api</artifactId>
+            <version>2.5</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>jsp-api</artifactId>
```
Comment (RomanDubovskyi) [inline pom.xml]: do we have jsp in project?

File: src/main/java/com/dev/cinema/config/AppConfig.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
+import org.springframework.orm.hibernate5.LocalSessionFactoryBean;
+
+@Configuration
+@ComponentScan(basePackages = {"com.dev.cinema"})
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/config/AppConfig.java]: it's better to set specific packages (service, dao, etc), with general package you may receive problems with two configuration files, if they overlap

File: src/main/java/com/dev/cinema/config/WebConfig.java [hunk 7]
```diff
@@ -0,0 +1,11 @@
+package com.dev.cinema.config;
+
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+
+@Configuration
+@ComponentScan(basePackages = {"com.dev.cinema"})
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/config/WebConfig.java]: same here, use controllers package

File: pom.xml (hunk 5 above)
Comment (vremere-a) [reply pom.xml]: no done

File: src/main/java/com/dev/cinema/config/AppConfig.java (hunk 6 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/config/AppConfig.java]: ok. done

File: src/main/java/com/dev/cinema/config/WebConfig.java (hunk 7 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/config/WebConfig.java]: done

File: pom.xml [hunk 8]
```diff
@@ -85,5 +129,4 @@
             </plugins>
         </pluginManagement>
     </build>
-
 </project>
```
Comment (RomanDubovskyi) [inline pom.xml]: missing empty line in the end of the file

File: src/main/java/com/dev/cinema/controllers/HelloController.java [hunk 9]
```diff
@@ -0,0 +1,14 @@
+package com.dev.cinema.controllers;
+
+import lombok.extern.log4j.Log4j;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@Log4j
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/controllers/HelloController.java]: we don't have logging here

File: pom.xml (hunk 8 above)
Comment (vremere-a) [reply pom.xml]: done

File: src/main/java/com/dev/cinema/controllers/HelloController.java (hunk 9 above)
Comment (vremere-a) [reply src/main/java/com/dev/cinema/controllers/HelloController.java]: ok. done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**